src/Compilers/Core/CommandLine/BuildProtocol.cs

```diff
@@ -426,7 +426,7 @@ public static ShutdownBuildResponse Create(BinaryReader reader)
         }
     }
 
-    file sealed class MismatchedVersionBuildResponse : BuildResponse
+    internal sealed class MismatchedVersionBuildResponse : BuildResponse
```

Comment:
Wouldn't other `file` types cause you the same problem?

Consider instead using `/pathmap` option to avoid leaking absolute paths into the output.

Reply:
Yes the file scope would cause this problem. As far as I know this is the only construct that causes different binary output when source paths differ. We use /pathmap only when output pdb symbols, which is not the case here.